test/metabase/test/data/interface.clj

```diff
@@ -440,6 +440,45 @@
      ~default-role
      (fn [] ~@body)))
 
+(defmulti create-user-with-pk!
+  "Creates a database user with the given public key"
+  {:added "0.55.0" :arglists '([driver details pk-user pub-key])}
+  dispatch-on-driver-with-test-extensions
+  :hierarchy #'driver/hierarchy)
```

Comment:
Should we add a default that along with these others throws errors? It feels a bit sneaky if you ask to create a user and silently you don't get one

Reply:
we can also add a feature like `:test/create-user` similar to `:test/cannot-destroy-db`